main.py

```diff
@@ -800,17 +801,18 @@ def handle_interface_link_down(self, interface):
         """
         Handler for interface link_down events
         """
+        log.info("Event handle_interface_link_down %s", interface)
         for evc in self.get_evcs_by_svc_level():
             with evc.lock:
-                log.info("Event handle_interface_link_down %s", interface)
                 evc.handle_interface_link_down(
                     interface
                 )
 
     @listen_to("kytos/topology.link_down")
     def on_link_down(self, event):
         """Change circuit when link is down or under_mantenance."""
-        self.handle_link_down(event)
+        with self._lock_handle_link_down:
+            self.handle_link_down(event)
```

Comment:
Why not make this dynamic single instead of using the lock? It doesn't make sense to have multiple threads queued to handle this if only one can do it at a time.

Reply:
Another part that to be aware that can be a bit expensive is this one here, `evc.setup_failover_path()` https://github.com/kytos-ng/mef_eline/blob/master/main.py#L892-L895, since it'll make a request to pathfinder, which could probably be deferred to be handled asynchronously in other event handlers.

Reply:
Good observation. Indeed `pool="dynamic_single"` is preferable for synchronous handlers.

Reply:
One points to keep in mind though is that there's still a `evc.sync()` DB write, that will be `O(len(evcs_affected_by_link))` in the worst case, each write on this collection here is typically 8-10 ms, so 200+ EVCs it can take 1+ sec, which might not be a major issue, since we do have amortized `kytos/topology.link_down|link_up`, but indeed it can be a bit wasteful in terms of IO doing synchronously here:

![20240222_135753](https://github.com/kytos-ng/mef_eline/assets/1010796/5651fdfa-a935-4b99-87d7-43e87efca4e8)

so, either maybe we also try to defer that `evc.sync()` or analyze what else can be done, I'm still trying to understand how the issue was resulting in EVCs get stuck.

Reply:
There's also `O(len(check_failover))` IO side effect on `handle_link_down` [when setting up the EVC failover](https://github.com/kytos-ng/mef_eline/blob/issue_405/main.py#L898-L901), which will make requests to `pathfinder`, so this one here if were to keep this single threaded we might also to deffer and compute asynchronously.

Reply:
Good suggestion @Ktmi . I've replaced the Lock with `dynamic_single` on commit a874260

Reply:
Looks good to me! Plus the deferred writes should help with resolving the DB write issue.